py/server/deephaven/table.py

```diff
@@ -97,6 +100,34 @@ class SearchDisplayMode(Enum):
     """Hide the search bar, regardless of user or system settings."""
 
 
+class AxisOption(Enum):
+    """An enum of axis options for data bars"""
+    PROPORTIONAL = _JAxisOption.Proportional
+    """Data bars fill the available space in the column."""
+    MIDDLE = _JAxisOption.Middle
+    """Zero line for data bars is centered in column."""
+    DIRECTIONAL = _JAxisOption.Directional
+    """Positive and negative bars face the same direction."""
+
+
+class ValuePlacementOption(Enum):
+    """An enum of value placement options for data bars"""
+    BESIDE = _JValuePlacementOption.Beside
+    """Beside the data bar."""
+    OVERLAP = _JValuePlacementOption.Overlap
+    """On top of the data bar."""
+    HIDE = _JValuePlacementOption.Hide
+    """Hide the value."""
+
+
+class DirectionOption(Enum):
+    """An enum of for horizontal axis orientation options for data bar"""
+    LTR = _JDirectionOption.LTR
+    """Left to right"""
+    RTL = _JDirectionOption.RTL
+    """Right to left"""
```

Comment:
I am OK with its being in the deephaven packageI but I'd like to propose that we create a webui module and move both format_data_bar() and the layout_hints() into it, which means both will not be methods on Table but functions that takes a Table as the 1st arg. This is a breaking change but will result in a clear separation between data ops and UI ops and table.py being less cluttered.  @mattrunyon @mofojed @chipkent your thoughts?

Reply:
I'm not sure I'd like the syntax when chaining them all together. For example, in Ethan's code above, he formats a bunch of data bars:
```python
data_bars = (
    students.format_data_bar(column="GPA", value_column="GPA")
    .format_data_bar(
        column="DirectionalAxis", value_column="DirectionalAxis", axis="directional"
    )
    .format_data_bar(column="MiddleAxis", value_column="MiddleAxis", axis="middle")
    .format_data_bar(
        column="OverlapPlacement",
        value_column="OverlapPlacement",
        value_placement="overlap",
    )
#    ... etc
)
```
With what you're proposing, you'd either have to nest all the tables, e.g.:
```python
data_bars = format_data_bars(format_data_bars(format_data_bars(students, ...), ...))
```
or use an intermediate variable (could be the same name), e.g.:
```python
data_bars = format_data_bars(students, ...)
data_bars = format_data_bars(students, ...)
... etc
```

Also would make it less discoverable (as you can see these functions currenty just from typing `t.` and seeing the autocomplete suggestions).

Could separate the format/UI functions as a separate interface but still implement it in the base table?

Reply:
I agree with @mofojed 's concerns.  As we have implemented things, formatting really is part of the table.  It isn't independent.

Reply:
To elaborate further on my final point from the previous comment - what would the implications be if we added the function to the `Table` prototype/metaclass when the Web UI module was loaded? e.g.
```python
def format_data_bar(self: Table, options: .... ):
    ...

def layout_hints(self: Table, options: ....):
    ...

Table.format_data_bar = format_data_bar
Table.layout_hints = layout_hints
```

This would separate the UI functions into their own module outside of the core table functionality, but the functions would still be available on the table itself (so could be written like my first snippet in previous comment) and available in autocomplete (for discoverability). Not sure where they would end up in our docs though.

Would something like that make sense? Or are there other cons I'm not thinking about?

Reply:
@mofojed I think what you proposed would work functionally but would mess up auto-complete and docs. Looking at the Java Table interface, as @chipkent mentioned, it is clear that we are treating UI ops as part of Table ops. So unless we are starting to separate the interfaces on the Java side, it doesn't make much sense to do that in Python. All things considered I am happy with the change as it stands now.

Reply:
Just to be clear, I checked doing the above and autocomplete seemed to pick it up correctly. We may want to discuss this further when we flesh out programmatic layouts and callbacks.